mne/preprocessing/xdawn.py

```diff
@@ -270,17 +272,17 @@ def fit(self, epochs, y=None):
         else:
             evokeds = dict()
             toeplitz = dict()
-            for eid in epochs.event_id:
-                evokeds[eid] = epochs[eid].average()
+            for eid in info['event_id']:
+                evokeds[eid] = np.mean(epochs_data[eid])
                 toeplitz[eid] = 1.0
         self.evokeds_ = evokeds
 
-        for eid in epochs.event_id:
-            data = np.dot(evokeds[eid].data, toeplitz[eid])
+        for eid in info['event_id']:
```

Comment:
How would we know about `event_id` ? Should it be passed in constructor? Same for `baseline`.
Also Jean told me we can know about `tmin` using `info` and `X.shape`. I couldn't figure out how.